master/internal/job/job.go

```diff
@@ -133,6 +141,213 @@ func (j *Jobs) jobQSnapshot(ctx *actor.Context, resourcePool string) (AQueue, er
 	return jobQ, nil
 }
 
+// figure out what changes are needed to move the job to the desired position.
+// Later we can support cross RP moving this way as well.
+func moveJobMessages(
+	jobs []*jobv1.Job,
+	target *jobv1.Job,
+	anchor *jobv1.Job,
+	anchorIdx int,
+	aheadOf bool,
```

Comment:
that'd be possible. I wanted to keep this closer to the user input for simpler unit testing. otherwise I think it'd work to swap the anchor with head or tail and remove the need for aheadOf